sched/queue/queue.go

```diff
@@ -12,6 +12,11 @@ type Queue interface {
 	// Enqueue enqueues a job and assigns it an ID. Errors may be
 	// an *InvalidJobRequest or a *CanNotScheduleNow
 	Enqueue(job sched.Job) (string, error)
+
+	Chan() chan WorkItem
+
+	Status(jobId string) (sched.JobStatus, error)
```

Comment:
I think we can take advantage of our (small) size to know that we can keep the Queue in-memory. So the durable log is for durability, and we can recover, but we know that our server will have the queue in-memory. So this should still be an in-memory operation for all reasonable implementations, and therefore very fast.

Does that make sense?